### DISCUSSION TITLE
LineTotalAmount does not take BasisQuantity into account?

### DISCUSSION THREAD

**Discussion by SW-Sylatex** [stephanstapel/ZUGFeRD-csharp#244] [Q&A] [answered]
Hello,

I am trying to add TradeLineItems with high Quantites like in the example of DIN EN 16931-1:
![Screenshot 2024-03-21 143805](https://github.com/stephanstapel/ZUGFeRD-csharp/assets/164358642/63f96c2e-9ad8-4f3a-970b-97575334f13a)

But instead the LineTotalAmount is calculated as "normal" (BilledQuantity * ChargeAmount)
instead of (as I understand): BilledQuantity / BasisQuantity * ChargeAmount

![grafik](https://github.com/stephanstapel/ZUGFeRD-csharp/assets/164358642/d67fe690-5dd2-45c5-913e-c86b35b699fa)
![grafik](https://github.com/stephanstapel/ZUGFeRD-csharp/assets/164358642/7bdf9f1b-c7f9-474f-ae65-88821f066fdd)

Hope someone can help/tell me if i am just doing something wrong, or if it is something else.

Best regards
Sven

**Comment by stephanstapel** [comment]
Yes, you are right.
This happens here:
https://github.com/stephanstapel/ZUGFeRD-csharp/blob/9253bd7434f771f07f1c8f51caba4da5d3300aac/ZUGFeRD/InvoiceDescriptor21Writer.cs#L413

and here accordingly:
https://github.com/stephanstapel/ZUGFeRD-csharp/blob/9253bd7434f771f07f1c8f51caba4da5d3300aac/ZUGFeRD/InvoiceDescriptor20Writer.cs#L353
https://github.com/stephanstapel/ZUGFeRD-csharp/blob/9253bd7434f771f07f1c8f51caba4da5d3300aac/ZUGFeRD/InvoiceDescriptor1Writer.cs#L576

Is it possible that you provide two test cases (with implicit BasisQuantity and with explicit BasisQuantity)? I'm happy to fix this bug in the core component then.

**Reply by SW-Sylatex** [reply]
Thank you for the fast response and sry for my late response...

I am sorry if i missunderstood your request of the two test cases or if i can not really help you, i am still an apprentice and my superior does not know/understand some of the "common" programming terms (specifically implicit and explicit in this case).
But i build a small test Programm with and without the BasisQuantity. 

Hope this helps somehow, if i should do/build something else please let me know.
I'm happy to help! (if i can)

Best regards,
Sven

[Zugferd-Test-CS.zip](https://github.com/stephanstapel/ZUGFeRD-csharp/files/14721193/Zugferd-Test-CS.zip)

**Comment by stephanstapel** [comment]
ok, no problem.
I added these two test cases:

https://github.com/stephanstapel/ZUGFeRD-csharp/commit/f3db5713eee46edcb1c654f7ff7f7748ba79dd6e

can you verify if this is what we are talking about?
I.e. setting the UnitQuantity to 10 so that 50 units will result in 5 billed quantity?

**Reply by SW-Sylatex** [reply]
> I.e. setting the UnitQuantity to 10 so that 50 units will result in 5 billed quantity?

Yes, what i expected would Happen is that in the calculation the billed quantity would be treated as 5 in that case. 

Thank you very much for your understanding

**Comment by stephanstapel** [comment, marked as answer]
ok, fixed:
https://github.com/stephanstapel/ZUGFeRD-csharp/commit/ec6c9b4bb76c390613561ab05df807b39ff7f91b
Please test with your code and close this discussion if everything works as expected
